### PR TITLE
Add confirmation popovers for release pets & mounts, rebirth, and fortify. Fixes #4104

### LINKED ISSUE #4104: Pop-Up Confirmation needed for Release Pets/Mount/both, Orb of Rebirth, Fortify Potion
A user who accidentally released all their mounts requested a pop-up confirmation for the Release Pets feature like there is for Rebirth. Probably a good idea.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/5063503-pop-up-confirmation-needed-for-release-pets-mount-both-orb-of-rebirth-fortify-potion?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by GeorgeSG**
Hi all,

This will be my first contribution :smile: I decided to take a shot at #4104 since it's marked as 'help needed' and it seems there's no progress. I've added confirmation popups for the actions (as it was discussed in [this PR](https://github.com/HabitRPG/habitrpg/pull/6100)).

I'm not sure if I should add some tests for this, as the SettingsCtrl is not heavily tested right now. Also, I'm not sure for the titles of the popovers ("Are you sure?") - made them like this for consistency, but if you think they should be more precise (e.g. "Release pets?", "Rebirth?", etc.) it's an easy change.

Also, the change in guildsCtrl is just formatting - while reading it I was annoyed that the indentation was not okay and decided to leave it in. 

Cheers,
George

Here are some screenshots:
rebirth
release
fortify

### CODE REVIEW

**Comment by GeorgeSG** [comment]
Here's my User ID: a187eae5-b3d2-479c-99c5-3e00b3857e24 :)

**Comment by crookedneighbor** [comment]
Tests are required.

**Comment by SabreCat** [comment]
Which is to say, part of how we're improving our test coverage--even in places where, as you note, it's not well covered yet--is to require that PRs come with tests. That way as we fix and update things, we simultaneously work down the test backlog!

**Comment by crookedneighbor** [comment]
Yes, that is what I meant to say (was on a phone when I commented).

**Comment by GeorgeSG** [comment]
:+1:  Okay, I'll add some tests tonight or tomorrow :)

**Comment by GeorgeSG** [comment]
I'm having issues running `npm test`. I've been talking with people in the _Aspiring Coders_ guild in-game to try and fix this, but here's a link to [a Gist describing my problem](https://gist.github.com/GeorgeSG/b3dfc2a32bc8f50af664) here, just in case someone else sees it and has a solution :) Thanks!

**Comment by GeorgeSG** [comment]
Added some tests. Please take a look :)

File: test/spec/controllers/settingsCtrlSpec.js [hunk 1]
```diff
@@ -83,4 +98,181 @@ describe('Settings Controller', function() {
...
+        expect(scope.popoverEl).to.exist;
+        sandbox.spy($.fn, 'popover');
+        scope.reroll(false);
+
+        $.fn.popover.should.have.been.calledWith('destroy');
+      });
+
+      it('doesn\'t call reroll when not confirmed', function() {
+        scope.reroll(false);
+        user.ops.reroll.should.not.have.been.called;
```
Comment (crookedneighbor) [inline test/spec/controllers/settingsCtrlSpec.js]: I prefer the expect syntax from `sinon-chai`:

```
expect(user.ops.reroll).to.not.be.called;
```

**Comment by GeorgeSG** [comment]
Refactored the tests to use the expect syntax. PTAL :)

**Comment by crookedneighbor** [comment]
Merged in with https://github.com/HabitRPG/habitrpg/commit/ea54ec607a51760c1a4c2cd1e69939572cab7fd8

This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by GeorgeSG** [comment]
@crookedneighbor, my user id is: a187eae5-b3d2-479c-99c5-3e00b3857e24
Thanks! :)

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

**Pull request closed without merging**